buzzmobile/tests/sense/test_inputer.py

```diff
@@ -0,0 +1,55 @@
+import rostest
+import unittest
+import rospy
+import time
+from std_msgs.msg import String
+import roslaunch
+
+PKG = 'buzzmobile'
+NAME = 'test_inputer'
+
+
+class TestInputer(unittest.TestCase):
+
+    def run_node_with_callback(self, callback):
+        rospy.init_node(NAME, anonymous=True)
+
+        rospy.Subscriber('destination', String, callback)
+        timeout = time.time() + 1.0
+
+        while time.time() < timeout:
+            time.sleep(0.1)
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+
+    def setUp(self):
+        self.success = False
+        node = roslaunch.core.Node('buzzmobile', 'inputer.py')
+        launch = roslaunch.scriptapi.ROSLaunch()
+        launch.start()
+        self.process = launch.launch(node)
+
+    def test_inputer_no_input(self):
+        def cb(data):
+            self.success = (data.data == '')
+
+        self.run_node_with_callback(cb)
+        assert self.success
+
+    def test_sanity(self):
+        def cb(data):
+            self.success = (data.data == b'success')
+
+        self.run_node_with_callback(cb)
+        assert not self.success
+
```

Comment:
I worked on that for a few hours, it comes down to some threading issues. Each ros node is in a different system thread, so normal mocking methods (`mock.patch` and co) can't run across threads. I also tried to do some stuff to try and fiddle with how the thread is created. For reference:

`roslaunch.scriptapi.ROSLaunch()` eventually runs some stuff from `roslaunch.nodeprocess.LocalProcess.start` which creates a `Popen` object, but doesn't set `stdin=subprocessing.PIP`. I monkeypatched that, but seemed to encounter some cross-thread timing errors (a deadlock, I think) when trying to send data to the other thread. Someone smarter than me may need to poke at it, or alternatively shoot out to StackOverflow to check up.